vars.tf

```diff
@@ -14,7 +28,5 @@ variable "event_pattern" {
 }
 
 locals {
-  lambda_names = toset(compact([for arn in var.targets : startswith(arn, "arn:aws:lambda")
-    ? element(split(":", arn), length(split(":", arn)) - 1)
-  : ""]))
+  lambda_names = [for arn in var.targets.lambda : element(split(":", arn), length(split(":", arn)) - 1)]
```

Comment:
Are you just trying to get at the name like `ministryOfMagic` here? Here's another way to do it (I think):
```tf
reverse(split(":", arn))[0]
```

Reply:
+1 just for the HP reference. My favorite highwing PR to date.

Reply:
Yes, the permission expects a name rather than an ARN